src/components/application_manager/src/commands/mobile/perform_interaction_request.cc

```diff
@@ -278,11 +279,14 @@ void PerformInteractionRequest::onTimeOut() {
       break;
     }
     case mobile_apis::InteractionMode::VR_ONLY: {
-      application_manager_.updateRequestTimeout(
-          connection_key(), correlation_id(), default_timeout());
+      LOG4CXX_DEBUG(logger_, "Interaction Mode: VR_ONLY");
+      unsubscribe_from_event(hmi_apis::FunctionID::UI_PerformInteraction);
+      DisablePerformInteraction();
+      CommandRequestImpl::onTimeOut();
```

Comment:
@LuxoftAKutsan, I can make it with if and it will looks like:
```
if ( interaction_mode_ != mobile_apis::InteractionMode::BOTH ||
      interaction_mode_ != mobile_apis::InteractionMode::VR_ONLY ||
      interaction_mode_ != mobile_apis::InteractionMode::MANUAL_ONLY ||) {
 return;
}

if ( interaction_mode_ != mobile_apis::InteractionMode::BOTH &&
      true == vr_response_recived_) {
     application_manager_.updateRequestTimeout(
            connection_key(), correlation_id(), default_timeout());
        return;
}

  unsubscribe_from_event(hmi_apis::FunctionID::UI_PerformInteraction);
  DisablePerformInteraction();
  CommandRequestImpl::onTimeOut();
```
And  note that type of PerformInteractionRequest won't log readable.